samples/quantum-approximation-optimization/qaoa-for-quadratic-unconstrained-binary-optimization.ipynb

```diff
@@ -0,0 +1,821 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "def func(param, *args):\n",
+    "    Q = args[0]\n",
+    "    C = quantum_loop(param)\n",
+    "    cost = get_cost(C, Q)\n",
+    "    return cost"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### QAOA for depth $p = 2$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now use a classical optimizer `minimize` in order to find the best variational parameters. This function takes as arguments `func`, the QUBO $Q$ and an initial point `x0` for the simplex in Nelder-Mead minimization. As the optimizer might get trapped in local minima, we repeat the optimization 20 times and select the parameters that yield the best approximation ratio.\n",
+    "\n",
+    "Note: the next cell may take up to 2 minutes to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "scores = []\n",
+    "params = []\n",
+    "for repetition in range(20):\n",
+    "    guess = {\n",
+    "        \"t\": np.random.uniform(1, 10, LAYERS),\n",
+    "        \"s\": np.random.uniform(1, 10, LAYERS),\n",
+    "    }\n",
+    "\n",
+    "    try:\n",
+    "        res = minimize(\n",
+    "            func,\n",
+    "            args=Q,\n",
+    "            x0=np.r_[guess[\"t\"], guess[\"s\"]],\n",
+    "            method=\"Nelder-Mead\",\n",
+    "            tol=1e-5,\n",
+    "            options={\"maxiter\": 10},\n",
+    "        )\n",
+    "        scores.append(res.fun)\n",
+    "        params.append(res.x)\n",
+    "    except Exception as e:\n",
+    "        pass"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now plot the sample that we woud obtain using the optimal variational parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "optimal_count_dict = quantum_loop(params[np.argmin(scores)])\n",
+    "plot_distribution(optimal_count_dict)"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "QAOA is capable of finding good variational parameters $\\tau$ and $t$. Now, sampling from this final state $|\\psi(t_{f})\\rangle$ will return both optimal strings with high probability."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "However, using QAOA to solve the problem is not the best idea; it's difficult to yield a >90% quality solution without going to high depths of the QAOA, implying that the growing closed-loop optimization can rapidly become expensive, with no guarantee of convergence. 
```

Comment:
Could you please split the string so it would be a valid JSON-array item?

Reply:
Oh, I think the issue comes from the suggestion I applied directly from Gitlab.

Should be fixed now.